src/postamble.js

```diff
@@ -280,7 +280,7 @@ function run(args) {
 #endif
 
 #if HAS_MAIN
-    if (Module['_main'] && shouldRunNow) Module['callMain'](args);
+    if (shouldRunNow) callMain(args);
```

Comment:
@kripken Removing this check for `Module['_main']` is causing an error in my library when I try to build with Emscripten 1.38.41: `Module['_main']` is undefined in my case.

My library does not have a `main` function, but the `HAS_MAIN` macro is set to true, because I build the library with the flag `-s MAIN_MODULE=2`. I compile with this flag so that I can use the `loadWebAssemblyModule` function to dynamically load WASM plugins that are compiled with `-s SIDE_MODULE=1`. I came up with this setup based on the [linking](https://github.com/emscripten-core/emscripten/wiki/Linking) and [standalone wasm](https://github.com/emscripten-core/emscripten/wiki/WebAssembly-Standalone) wiki pages.

Was removing this condition intentional? If so, it sounds like I should not be using `MAIN_MODULE=2` in my case. In that case, how do I ensure that the `loadWebAssemblyModule` function gets included?

Reply:
Thanks @maxbrunsfeld! Looks like that's a regression from this PR, our test suite had no coverage for a main module without a `main()` function. Fix in #9186

Reply:
⚡️ Thanks so much for the quick reply!